### PR TITLE
First commit of Nav branch

### PR DESCRIPTION
**Pull request by collinnolen**

### CODE REVIEW

File: Assets/Scripts/Navigation/KeyBoardManager.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	void Start () {
+		
+	}
+	
+	// Update is called once per frame
+	void Update () {
+		ArrowKeys();
+	}
+
+	private void ArrowKeys(){
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/KeyBoardManager.cs]: Maybe leave this as a TODO comment, but we should use a command pattern (http://gameprogrammingpatterns.com/command.html) to handle inputs instead of having a keypress directly transform the camera. It's probably out of scope for now, but I think that will be important to implement later

File: Assets/Scripts/Navigation/KeyBoardManager.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class KeyBoardManager : MonoBehaviour {
+
+	public int cameraMovespeed = 5;
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/KeyBoardManager.cs]: any reason for making this public? also, maybe this should be a static variable?

File: Assets/Scripts/Navigation/Map.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Map : MonoBehaviour {
+
+	public GameObject hexPrefab;
+
+	int width = 20;
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/Map.cs]: should width, height, xOffset and yOffset be static?

File: Assets/Scripts/Navigation/Map.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+				hex_go.tag = "Water";
+
+				hex_go.transform.SetParent(this.transform);
+			}
+		}
+	}
+
+	private void HexMap(){
+		Debug.Log("hex");
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/Map.cs]: Does unity have an equivalent to "throw"? Let's just throw an error here instead of outputting text

File: Assets/Scripts/Navigation/MouseManager.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MouseManager : MonoBehaviour {
+
+	//Fields Used for Camera
+	public int cameraEdgeBoundary = 100;
+ 	public int cameraMovespeed = 2;
+ 
+ 	private int theScreenWidth;
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/MouseManager.cs]: can we remove "the" from the variable names?

File: Assets/Scripts/Navigation/MouseManager.cs [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+ 	private int theScreenHeight;
+
+	// Use this for initialization
+	void Start () {
+		theScreenWidth = Screen.width;
+     	theScreenHeight = Screen.height;
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/MouseManager.cs]: spacing needs to be fixed

File: Assets/Scripts/Navigation/MouseManager.cs [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+			//Raycast hit something.
+
+			if(Input.GetMouseButtonDown(0)){
+				//Left Click
+				Debug.Log("Hit: " + hit.collider.name);
+			}
+		}
+	}
+
+	private void MoveCamera(){
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/MouseManager.cs]: tabbing seems a bit weird in this method, see if you can fix that

File: Assets/Scripts/Navigation/Map.cs [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+	public GameObject hexPrefab;
+
+	int width = 20;
+	int height = 20;
+
+	float xOffset = 0.96f;
+	float yOffset = 0.71f;
+
+	public string mapShape = "hex";
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/Map.cs]: can we just make it "HEX" so we don't have to call ToUpper() on it in BuildMap()?

File: Assets/Scripts/Navigation/Player.cs [hunk 9]
```diff
@@ -1,208 +1,25 @@
...
-        }
-    }
-    // Update is called once per frame
-    void Update () {
-        if(tiles == null)
-        {
-            tiles = grid.getTiles();
-        }
-        status.text = "X: " + x + "  Y: " + y;
+	private void Battle(){
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/Player.cs]: Can you add a quick comment of what this method should do, something akin to javadocs? Like "Transition player into battle"

File: Assets/Scripts/Navigation/Player.cs [hunk 10]
```diff
@@ -1,208 +1,25 @@
...
+		//Todo
 	}
+
+	public void CantMove(){
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/Player.cs]: This seems like a weird method to have, what is this method supposed to do?

File: Assets/Scripts/Navigation/NavController.cs [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+		if(instance == null)
+			instance = this;
+		else if(instance != this)
+			Destroy(gameObject);
+
+		DontDestroyOnLoad(gameObject);	
+
+		map = Instantiate(map, Vector3.zero, Quaternion.identity);
+		map.mapShape = "rect";
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/NavController.cs]: add a comment here saying this is "temporary code" for testing, as this line is not directly related to the functionality

File: Assets/Scripts/Navigation/Map.cs [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+	}
+
+	void Update(){
+
+	}
+
+	private void BuildMap(){
+		switch(mapShape.ToUpper()){
+			//Builds a hex shaped map
+			case "HEX"	: //Todo
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/Map.cs]: Instead of a Todo, just have it call the HexMap() method, where it should throw an error

File: Assets/Scripts/Navigation/KeyBoardManager.cs [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+
+	// Use this for initialization
+	void Start () {
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/KeyBoardManager.cs]: Remove the " " before ()

File: Assets/Scripts/Navigation/KeyBoardManager.cs [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+		
+	}
+	
+	// Update is called once per frame
+	void Update () {
+		ArrowKeys();
+	}
+
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/KeyBoardManager.cs]: Remove the " " before ()

File: Assets/Scripts/Navigation/MouseManager.cs (hunk 6 above)
Comment (artsiriwatt) [reply Assets/Scripts/Navigation/MouseManager.cs]: Remove the " " before ()

File: Assets/Scripts/Navigation/MouseManager.cs [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+ 	private int theScreenHeight;
+
+	// Use this for initialization
+	void Start () {
+		theScreenWidth = Screen.width;
+     	theScreenHeight = Screen.height;
+	}
+	
+	// Update is called once per frame
+	void Update () {
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/MouseManager.cs]: Remove the " " before ()

File: Assets/Scripts/Navigation/Player.cs [hunk 16]
```diff
@@ -1,208 +1,25 @@
...
-
-    // Use this for initialization
-    void Start () {
-        pos.position = new Vector3(0, 0, -5);
-        x = 0;
-        y = 0;
-        right = false;
+	
+	// Use this for initialization
+	void Start () {
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/Player.cs]: Remove the " " before ()

File: Assets/Scripts/Navigation/Player.cs [hunk 17]
```diff
@@ -1,208 +1,25 @@
...
-            {
-                y++;
-                right = true;
-                pos.position = new Vector3(pos.position.x + .5f, pos.position.y + 0.866025404f, -5);
-                moveStatus.text = "";
-            }
-        }
-    }
+	// Update is called once per frame
+	void Update () {
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/Player.cs]: Remove the " " before ()

File: Assets/Scripts/Navigation/KeyBoardManager.cs [hunk 18]
```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class KeyBoardManager : MonoBehaviour {
+
+	public int cameraMovespeed = 5;
+
+	// Use this for initialization
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/KeyBoardManager.cs]: Let's also remove these comments for Start() and Update(), they don't provide any information we don't know

File: Assets/Scripts/Navigation/Player.cs [hunk 19]
```diff
@@ -1,208 +1,25 @@
...
+		//Todo
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/Player.cs]: See the comment about TODOs I left for Jeremy: https://github.com/jgayewu/pirategame/pull/11#discussion_r179009928

File: Assets/Scripts/Navigation/Map.cs [hunk 20]
```diff
@@ -0,0 +1,151 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Map : MonoBehaviour {
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/Map.cs]: there are some spacing issues in this class

File: Assets/Scripts/Navigation/Hex.cs [hunk 21]
```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Hex : MonoBehaviour {
+
+    public Map map;
+    public List<Hex> neighboors;
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/Hex.cs]: should be "neighbors"

File: Assets/Scripts/Navigation/Hex.cs [hunk 22]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        this.GetComponent<SpriteRenderer>().sprite = sprite;
+    }
+
+    void OnMouseUp()
+    {
+        map.player.GetComponent<Player>().MovePlayer(this.x, this.y);
+    }
+
+    public static int Comparison(Hex hex1, Hex hex2)
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/Hex.cs]: any reason this is returns an int and not a bool?

File: Assets/Scripts/Navigation/MapShapeFactory.cs [hunk 23]
```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public static class MapShapeFactory{
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/MapShapeFactory.cs]: Cleanup spacing issues

File: Assets/Scripts/Navigation/MouseManager.cs [hunk 24]
```diff
@@ -0,0 +1,64 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MouseManager : MonoBehaviour {
```
Comment (artsiriwatt) [inline Assets/Scripts/Navigation/MouseManager.cs]: Spacing issues

Comment (collinnolen) [reply Assets/Scripts/Navigation/MouseManager.cs]: sd

**Comment by artsiriwatt** [comment]
merged via #17
